migration-engine/core/src/commands/create_migration.rs

```diff
@@ -33,11 +35,56 @@ impl<'a> MigrationCommand for CreateMigrationCommand {
 
     type Output = CreateMigrationOutput;
 
-    async fn execute<C, D>(_input: &Self::Input, _engine: &MigrationEngine<C, D>) -> super::CommandResult<Self::Output>
+    async fn execute<C, D>(input: &Self::Input, engine: &MigrationEngine<C, D>) -> super::CommandResult<Self::Output>
     where
         C: migration_connector::MigrationConnector<DatabaseMigration = D>,
         D: migration_connector::DatabaseMigrationMarker + Send + Sync + 'static,
     {
-        todo!("createMigration command")
+        let database_migration_inferrer = engine.connector().database_migration_inferrer();
+        let applier = engine.connector().database_migration_step_applier();
+        let checker = engine.connector().destructive_change_checker();
+
+        // Infer the migration.
+        let previous_migrations = migration_connector::list_migrations(&Path::new(&input.migrations_directory_path))
+            .map_err(|_| {
+                CommandError::Generic(anyhow::anyhow!(
+                    "Failed to read migrations from the migrations directory."
+                ))
+            })?;
+        let target_schema = parse_datamodel(&input.prisma_schema)?;
+
+        let migration = database_migration_inferrer
+            .infer_next_migration(&previous_migrations, &target_schema)
+            .await?;
+
+        if migration.is_empty() && !input.draft {
+            return Ok(CreateMigrationOutput {
+                generated_migration_name: None,
+            });
+        }
+
+        let destructive_change_diagnostics = checker.pure_check(&migration);
```

Comment:
TODO: issue to link the destructive change checks better with the steps (needs product input)